boards/arm/rp2040/common/src/rp2040_st7789.c

```diff
@@ -0,0 +1,124 @@
+/****************************************************************************
+ * boards/arm/rp2040/common/src/rp2040_st7789.c
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.  The
+ * ASF licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <nuttx/config.h>
+
+#include <stdio.h>
+#include <stdbool.h>
+#include <debug.h>
+#include <errno.h>
+
+#include <nuttx/arch.h>
+#include <nuttx/board.h>
+#include <nuttx/spi/spi.h>
+#include <nuttx/lcd/lcd.h>
+#include <nuttx/lcd/st7789.h>
+
+#include "rp2040_spi.h"
+#include "rp2040_gpio.h"
+
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+
+#define LCD_SPI_PORTNO 0
+
+/****************************************************************************
+ * Private Data
+ ****************************************************************************/
+
+static struct spi_dev_s *g_spidev;
+static struct lcd_dev_s *g_lcd = NULL;
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name:  board_lcd_initialize
+ *
+ * Description:
+ *   Initialize the LCD video hardware.  The initial state of the LCD is
+ *   fully initialized, display memory cleared, and the LCD ready to use, but
+ *   with the power setting at 0 (full off).
+ *
+ ****************************************************************************/
+
+int board_lcd_initialize(void)
+{
+  g_spidev = rp2040_spibus_initialize(LCD_SPI_PORTNO);
```

Comment:
LCD_SPI_PORTNO shouldn't be hardcoded, pass it as a function parameter!

Reply:
Thanks for comment. There are some reasons for it.
- board_lcd_initialize() is common function called from lcd_framebuffer.c and it is not designed to pass a parameter. Another existing lcd support code also takes similar way.
- The "Pico Display Pack" designed for Raspberry Pi Pico has fixed pin layout. It can use only SPI ch 0.

Reply:
> Thanks for comment. There are some reasons for it.
> 
>     * board_lcd_initialize() is common function called from lcd_framebuffer.c and it is not designed to pass a parameter. Another existing lcd support code also takes similar way.
> 
>     * The "Pico Display Pack" designed for Raspberry Pi Pico has fixed pin layout. It can use only SPI ch 0.

Ah ok! Thank you for this explanation!